File: agent/bindings/csharp/Uri.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class Uri
  {
    [JsonProperty(PropertyName = "scheme")]
    public string Scheme { get; set; }
    [JsonProperty(PropertyName = "authority")]
    public string Authority { get; set; }
    [JsonProperty(PropertyName = "path")]
    public string Path { get; set; }
    [JsonProperty(PropertyName = "query")]
    public string Query { get; set; }
    [JsonProperty(PropertyName = "fragment")]
    public string Fragment { get; set; }
    [JsonProperty(PropertyName = "fsPath")]
    public string FsPath { get; set; }
  }
}

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Uri.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class Uri(
  val scheme: String,
  val authority: String,
  val path: String,
  val query: String,
  val fragment: String,
  val fsPath: String,
)

